CRM/Utils/REST.php

```diff
@@ -546,6 +546,14 @@ public static function ajaxJson() {
       );
       CRM_Utils_JSON::output($error);
     }
+    // FIXME: Shouldn't the X-Forwarded-Proto check be part of CRM_Utils_System::isSSL()?
```

Comment:
@seamuslee001 the problem with that approach is that you end up loading the bootstrap multiple times in some cases. I patched the three calls to redirectToSSL to use the above unconditional code, and it all seemed to work, but was greeted with this when I loaded a Civi page:

`Notice: Constant DRUPAL_ROOT already defined in CRM_Utils_System_Drupal->loadBootStrap() (line 477 of /srv/www/d7-master/sites/all/modules/civicrm/CRM/Utils/System/Drupal.php).`